tests/test_load_dump.py

```diff
@@ -599,3 +600,25 @@ def test_float_subclass(self):
             rate=decimal.Decimal("0.000012345"),
         )
         self.check_dump_load_roundtrip(b.resolve())
+
+    def test_json_infinities_get_stringified(self):
+        re_inf = re.compile(r'[^"]\bInfinity\b[^"]')
+        assert re_inf.search("start_time: Infinity,") is not None
+        assert re_inf.search("start_time: Infinity}") is not None
+        assert re_inf.search('start_time: "Infinity",') is None
+        assert re_inf.search('start_time: "Infinity"}') is None
+
+        re_inf_str = re.compile(r'"Infinity"')
+        assert re_inf_str.search("start_time: Infinity,") is None
+        assert re_inf_str.search("start_time: Infinity}") is None
+        assert re_inf_str.search('start_time: "Infinity",') is not None
+        assert re_inf_str.search('start_time: "Infinity"}') is not None
+
+        b = demes.Builder()
+        b.add_deme("a", epochs=[dict(start_size=1)])
+        b.add_deme("b", epochs=[dict(start_size=1)])
+        b.add_migration(source="a", dest="b", rate=1e-4)
+        g = b.resolve()
+        json_str = demes.dumps(g, format="json", simplified=False)
+        assert re_inf.search(json_str) is None
```

Comment:
Not clear why you're faffing around with regexes here - much simpler (and more obvious) to test the known structure of the data dictionary